### PR TITLE
526 allow the currently active sync setting to be deleted

### LINKED ISSUE #526: Allow the currently active sync setting to be deleted
**Is your feature request related to a problem? Please describe.**
Currently I can only 'Add new credentials' or 'Edit' existing ones. I would like to be able to delete existing settings, especially since they (potentially) contain PAT for GitHub and other sensitive information. I'm not even able to fill in 'garbage info' afterwards because the plugin is then unable to connect to GitHub - a nice feature when you're adding a synch config the first time, less nice when you are trying anonymize data after initial add.

**Describe the solution you'd like**
A delete button next to the 'Edit' option which permanently deletes the configuration would be ideal

### PR DESCRIPTION
**Pull request by swordEdge**
Fixes #526 

- [x] Functional review: @six7

### CODE REVIEW

**Comment by six7** [comment]
@swordEdge bit of functional feedback:

Remove the `Delete` button. Instead use what we already have: `Delete local credentials` beneath the repo. Sorry for causing extra work as this wasn't in the issue description. That way we don't introduce another state to that component.
image

File: src/app/components/Button/Button.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { StyledButton } from './StyledButton';
 export interface ButtonProps {
   type?: 'button' | 'submit';
   form?: string
-  variant: 'secondary' | 'primary' | 'ghost';
```
Comment (LiamMartens) [inline src/app/components/Button/Button.tsx:9]: Space can be removed

**Review by LiamMartens — APPROVED** [review]

**Comment by six7** [comment]
@swordEdge could you add a test for that file so coverage passes?

File: src/app/components/StorageItem.tsx [hunk 2]
```diff
@@ -23,14 +24,22 @@ const StorageItem = ({ item, onEdit }: Props) => {
   const branch = isGitProvider(item) ? item.branch : null;
 
   const { restoreStoredProvider, deleteProvider } = useRemoteTokens();
+  const { confirm } = useConfirm();
+
+  const askUserIfDelete = React.useCallback(async () => {
+    const shouldDelete = await confirm({
+      text: 'Do you really want to delete a sync setting?',
```
Comment (six7) [inline src/app/components/StorageItem.tsx:31]:
```suggestion
      text: 'Do you really want to delete this sync setting?',
```

**Pull request merged**